File: Solutions/58-LengthofLastWord/58-LengthofLastWord.py
# Last updated: 6/5/2026, 6:17:09 PM
1class Solution(object):
2    # @param grid, a list of lists of integers
3    # @return an integer
4    def minPathSum(self, grid):
5        sum = list(grid[0])
6        for j in xrange(1, len(grid[0])):
7            sum[j] = sum[j - 1] + grid[0][j]
8
9        for i in xrange(1, len(grid)):
10            sum[0] += grid[i][0]
11            for j in xrange(1, len(grid[0])):
12                sum[j] = min(sum[j - 1], sum[j]) + grid[i][j]
13
14        return sum[-1]

File: Solutions/58-LengthofLastWord/58-LengthofLastWord.cpp
// Last updated: 6/5/2026, 6:14:29 PM
1class Solution {
2    public:
3        unordered_map<int, unordered_map<int, int> > hash;
4        int uniquePaths(int m, int n) {
5            if(m == 0 || n == 0)
6                return 0;
7            if(m == 1 || n == 1)
8                return 1;
9            if(hash.find(m) != hash.end() && hash[m].find(n) != hash[m].end())
10                return hash[m][n];
11            return hash[m][n] = uniquePaths(m - 1, n) + uniquePaths(m, n - 1);
12        }
13};